src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -2670,5 +2671,260 @@ private interface INegator<T> where T : struct
+                        nuint secondOffset = firstOffset + numElements;
+                        nuint lastOffset = length - (i * numElements * 2) - numElements;
+                        nuint secondLastOffset = lastOffset - numElements;
+
+                        // Load in values from beginning and end of the array.
+                        Vector128<T> tempFirst = Vector128.LoadUnsafe(ref elements, firstOffset);
+                        Vector128<T> tempSecond = Vector128.LoadUnsafe(ref elements, secondOffset);
+                        Vector128<T> tempLast = Vector128.LoadUnsafe(ref elements, lastOffset);
+                        Vector128<T> tempSecondLast = Vector128.LoadUnsafe(ref elements, secondLastOffset);
+
+                        // Shuffle to reverse each vector
+                        if (Unsafe.SizeOf<T>() == sizeof(byte))
+                        {
+                            //     +-------------------------------------------------------------------------------+
+                            //     | A1 | B1 | C1 | D1 | E1 | F1 | G1 | H1 | I1 | J1 | K1 | L1 | M1 | N1 | O1 | P1 |
+                            //     +-------------------------------------------------------------------------------+
+                            //     | A2 | B2 | C2 | D2 | E2 | F2 | G2 | H2 | I2 | J2 | K2 | L2 | M2 | N2 | O2 | P2 |
+                            //     +-------------------------------------------------------------------------------+
+                            //         Shuffle --->
+                            //     +-------------------------------------------------------------------------------+
+                            //     | P2 | O2 | N2 | M2 | L2 | K2 | J2 | I2 | H2 | G2 | F2 | E2 | D2 | C2 | B2 | A2 |
+                            //     +-------------------------------------------------------------------------------+
+                            //     | P1 | O1 | N1 | M1 | L1 | K1 | J1 | I1 | H1 | G1 | F1 | E1 | D1 | C1 | B1 | A1 |
+                            //     +-------------------------------------------------------------------------------+
+                            tempFirst = Vector128.Shuffle(tempFirst.As<T, byte>(), Vector128.Create(
+                                (byte)15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0)).As<byte, T>();
+                            tempSecond = Vector128.Shuffle(tempSecond.As<T, byte>(), Vector128.Create(
+                                (byte)15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0)).As<byte, T>();
+                            tempLast = Vector128.Shuffle(tempLast.As<T, byte>(), Vector128.Create(
+                                (byte)15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0)).As<byte, T>();
+                            tempSecondLast = Vector128.Shuffle(tempSecondLast.As<T, byte>(), Vector128.Create(
+                                (byte)15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0)).As<byte, T>();
```

Comment:
Is this really better? We only get 2 shuffle ports on most modern hardware so we were already saturating the loop with two shuffles. The smaller loop should also have meant everything stays in the micro-op cache and a single decode window.